web/ui/react-app/src/Checkbox.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { FC, useState, HTMLProps, memo } from 'react';
+import { FontAwesomeIcon, FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
+
+interface CheckboxProps extends Omit<HTMLProps<HTMLSpanElement>, 'size' | 'onClick'> {
+  onToggle: (selected: boolean) => void
+}
+
+const Checkbox: FC<CheckboxProps & Pick<FontAwesomeIconProps, 'size'>> = props => {
+  const { children, onToggle, size = 'sm', checked = false, ...rest } = props;
+  const [selected, setSelected] = useState(checked);
+  const handleClick = () => {
+    setSelected(!selected);
+    onToggle && onToggle(!selected);
+  }
+  return (
+    <span {...rest} onClick={handleClick} className={`checkbox${(selected ? ' checked' : '')}`}>
+      <FontAwesomeIcon size={size} icon={['far', (selected ? 'check-square' : 'square')]}/>
+      <span className="checkbox-label">{children}</span>
+    </span>
+  )
+};
+
+export default memo(Checkbox);
```

Comment:
Nit: please configure your editor to always end files with a terminating newline character (e.g. `"files.insertFinalNewline": true` in VS Code).

Reply:
Thanks, I'll check that setting

Reply:
prettier is a pretty decent tool to enforce consistent style in es code (gofmt for js), you can hook it into eslint via https://prettier.io/docs/en/integrating-with-linters.html
there are also plugins for editors so it's run on file save

Reply:
Thanks, I filed https://github.com/prometheus/prometheus/issues/6199 for someone to improve the linter situation :)